app/src/main/java/com/babylon/wallet/android/presentation/transfer/TransferViewModel.kt

```diff
@@ -301,6 +332,26 @@ class TransferViewModel @Inject constructor(
             ).toPersistentList()
         ).withCheckedBalances()
 
+        fun withUpdatedDepositRules(): State {
+            val targetAccounts = targetAccounts.map { targetAccount ->
+                val accountDepositResourceRule = accountDepositResourceRulesSet?.find { it.accountAddress == targetAccount.address }
+                targetAccount.updateAssets { assets ->
+                    assets.map { asset ->
+                        when (asset) {
+                            is SpendingAsset.Fungible -> asset.copy(
+                                canDeposit = accountDepositResourceRule?.canDeposit(asset.resourceAddress) ?: true
+                            )
+
+                            is SpendingAsset.NFT -> asset.copy(
+                                canDeposit = accountDepositResourceRule?.canDeposit(asset.resourceAddress) ?: true
```

Comment:
`val canDeposit = accountDepositResourceRule?.canDeposit(asset.resourceAddress) ?: true`

This can be written once above the `when` statement.